File: python/problema_1_3.py
'''
El programa, dado el nombre de un dinosaurio, su peso expresado en toneladas y
su longitud expresada en pies; escribe el nombre del dinosuario, su peso y longitud
expresadas en kilogramos y metros, respectivamente.
'''

def convertir_unidades(peso, longitud):

    peso_kilogramos = peso * 1000
    # Una tonelada es igual a mil kilogramos

    longitud_metros = longitud * 0.3047
    # Un pie es igual a 0.3047 metros

    return peso_kilogramos, longitud_metros

def main():
    
    try:

        nombre = input("Nombre del dinosaurio: ")

        peso = float(input("Peso en toneladas: "))

        longitud = float(input("Longitud en pies: "))

        if peso <= 0 or longitud <= 0:

            print("Error: el peso y la longitud deben ser mayores que cero.")
            return

    except ValueError as e:
        # Se produce cuando una función recibe un argumento 
        # con un valor que no es válido para esa función.

        print("Error: por favor, ingresa un valor válido.")
        return
    
    peso_kilogramos, longitud_metros = convertir_unidades(peso, longitud)

    print(f"El dinosaurio {nombre}")

    print(f"pesa {peso_kilogramos:.2f} kilogramos y")

    print(f"mide {longitud_metros:.2f} metros")

if __name__ == "__main__":

    main()

File: java/unidad_01/Problema_1_3/src/Problema_1_3.java
import java.util.Scanner;
import java.util.InputMismatchException;

/*
 * 
 * El programa, dado el nombre de un dinosaurio, su peso expresado en toneladas y
 * su longitud expresada en pies; escribe el nombre del dinosuario, su peso y longitud
 * expresadas en kilogramos y metros, respectivamente.
 *
 * @author Angel P.
 * 
 */

public class Problema_1_3 {

    public static float[] convertirUnidades(float peso, float longitud) {

        float pesoKilogramos = peso * 1000;
        // Una tonelada es igual a mil kilogramos

        float longitudMetros = longitud * 0.3047f;
        // Un pie es igual a 0.3047 metros

        return new float[] { pesoKilogramos, longitudMetros };
    }

    public static void main(String[] args) {

        String nombre;

        float peso, longitud;

        try (Scanner scanner = new Scanner(System.in)) {
            /*
             * Es una buena práctica cerrar los recursos adecuadamente después de usarlos
             * para liberar cualquier recurso asociado, como archivos abiertos, conexiones
             * de red o lectores de entrada.
             */

            System.out.println("Nombre del dinosaurio: ");
            nombre = scanner.nextLine();

            System.out.println("Peso en toneladas: ");
            peso = scanner.nextFloat();

            System.out.println("Longitud en pies: ");
            longitud = scanner.nextFloat();

            if (peso <= 0 || longitud <= 0) {

                System.out.println("Error: el peso y la longitud deben ser mayores que cero.");
                return;
            }
        } catch (InputMismatchException e) {
            // Se produce cuando el tipo de dato ingresado por el usuario no coincide con el
            // esperado.

            System.out.println("Error: por favor, ingresa un valor válido.");
            return;
        }
        float[] unidades = convertirUnidades(peso, longitud);

        System.out.printf("El dinosaurio %s %n", nombre);

        System.out.printf("pesa %.2f kilogramos y %n", unidades[0]);

        System.out.printf("mide %.2f metros", unidades[1]);
    }
}